### PR TITLE
[repo] docs: revamp getting started docs

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #733 #1126 #732  (issue number)

Adding: @corinagum, @Carter425 as contributors since I rephrased and relocated docs written by them.

## Details

**Added the following sections**
- Overview - Introduction to the getting started docs
- Quickstart - Step by step instructions to get the echo bot set up.
- Samples - Overview of all the samples in the sdk and supported language.
- Migration (Overview, JS, C#) - How to migrate a BF SDK app to Teams AI app.

**Tweaked the following sections**
- C#, JS package readme's migration section has been merged into the getting started docs
- Samples folder readme now points to the getting started docs "sample" section.

**Removed the following sections from the existing getting started docs**
- dotnet/*
- js/*
- 00.PROMPTs - 05.TURNS
- README.md (replaced by 00.OVERVIEW.md)

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (we use [TypeDoc](https://typedoc.org/) to document our code)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: getting-started/00.OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+- [**00.OVERVIEW**](./00.OVERVIEW.md)
```
Comment (aacebo) [inline getting-started/00.OVERVIEW.md:1]: I personally feel like a title should always begin the document, seeing the table of contents first without any context feels odd to me.

**Comment by aacebo** [comment]
I see that we are removing the docs that introduce/explain to developers our different concepts in the SDK, is this somewhere else now? For many people using the SDK it will be their first time interacting with AI so I think its important for us to explain what our different concepts are and how they contribute to the greater whole.

File: js/samples/README.md [hunk 2]
```diff
@@ -1,114 +1,4 @@
 # Sample Library
 
-The following samples highlight key capabilities of the Teams AI SDK.
+You can find the sample library in the [getting started docs](../../getting-started/02.SAMPLES.md).
```
Comment (corinagum) [inline js/samples/README.md:3]: Do we need to update / delete the equivalen content in the root readme about capabilities as well? Or are they different enough to keep both?

image

File: getting-started/00.OVERVIEW.md (hunk 1 above)
Comment (corinagum) [reply getting-started/00.OVERVIEW.md:1]: This also matches msdocs rules, to have H1 at the very top of the document.

**Review by corinagum — DISMISSED** [review]
LGTM - approving, but please address what we discussed in call:
1. I think having 'under construction' sections for content that is out of date is okay, but let's not 100% delete stuff like turns and actions, etc. Just in case it takes a while to circle back to that content, I'd rather not lose it. 
2. Asking Carter to look through everything in a 'new to the repo' context for sanity / organization comments.
3. Matching JS directions to DOTNET re: directions on running the sample in one location.

File: getting-started/00.OVERVIEW.md (hunk 1 above)
Comment (singhk97) [reply getting-started/00.OVERVIEW.md:4]: fixed

**Comment by singhk97** [comment]
> I see that we are removing the docs that introduce/explain to developers our different concepts in the SDK, is this somewhere else now? For many people using the SDK it will be their first time interacting with AI so I think its important for us to explain what our different concepts are and how they contribute to the greater whole.

Agreed. We will continue to add docs in the coming weeks.

File: js/samples/README.md (hunk 2 above)
Comment (singhk97) [reply js/samples/README.md]: The capabilities can be a sample repo specific. The getting start "sample" docs is to give a high level overview.

**Comment by singhk97** [comment]
> LGTM - approving, but please address what we discussed in call:
> 
> 1. I think having 'under construction' sections for content that is out of date is okay, but let's not 100% delete stuff like turns and actions, etc. Just in case it takes a while to circle back to that content, I'd rather not lose it.
> 2. Asking Carter to look through everything in a 'new to the repo' context for sanity / organization comments.
> 3. Matching JS directions to DOTNET re: directions on running the sample in one location.

1. I've added `Turns` guide. As for actions...etc and concepts related to the AI system I would need more time to carefully rewrite. Even though it's deleted, it still lives in the commit history so we can always go back and retrieve it.
2. I like this idea. Will ask him.
3. I didn't match is 100% per se but made it very similar.

**Review by aacebo — APPROVED** [review]

**Pull request merged**